src/reflect/scala/reflect/internal/util/BooleanContainer.java

```diff
@@ -0,0 +1,61 @@
+package scala.reflect.internal.util;
+
+import scala.reflect.internal.util.AlmostFinalValue;
+import java.lang.invoke.MethodHandle;
+
+/**
+ * Represents a container with a boolean value that tells the compiler whether
+ * an option is enabled or not. This class is used for configuration purposes
+ * (see scala.reflect.internal.util.Statistics).
+ * 
+ * Its implementation delegates to {@link scala.reflect.internal.util.AlmostFinalValue},
+ * which helps performance (see docs to find out why).
+ */
+public class BooleanContainer {
+  private final boolean value;
+
+  public BooleanContainer(boolean value) {
+    this.value = value;
+  }
+  
+  public boolean isEnabled() {
+    return value;
+  }
+
+  protected final static class TrueBooleanContainer extends BooleanContainer {
+    TrueBooleanContainer() {
+        super(true);
+    }
+  }
+
+  protected final static class FalseBooleanContainer extends BooleanContainer {
+    FalseBooleanContainer() {
+        super(false);
+    }
+  }
+  
+  private static final AlmostFinalValue<BooleanContainer> DEFAULT_VALUE = new AlmostFinalValue<BooleanContainer>() {
+    @Override
+    protected BooleanContainer initialValue() {
+        return new FalseBooleanContainer();
+    }
+  };
+
+  private static final MethodHandle DEFAULT_VALUE_GETTER = DEFAULT_VALUE.createGetter();
+  
+  public static BooleanContainer getDefault() {
```

Comment:
Should this actually be named `isStatisticsEnabled`? AFAICT it can't be used for more than one purpose, so it seems strange to leave it here.

If my assumptions are true, I'd suggest to move `getDefault` and the static fiekds it depends on to a class named `StatisticsStatics` or somesuch.

Reply:
Hmm, the naming is generic is because I thought that this `BooleanContainer` could be used in other scenarios (that require an almost final boolean), and so didn't want to hardcode it. There's in theory nothing tied to statistics on this class on purpose.
  
I'm happy to follow your advice if you think that this genericity will not be worth it in the future... but please confirm :smile:.

Reply:
Yes, please do. IMO this doesn't make any more sense than exposing `var default: Boolean` here :)